packages/cozy-pouch-link/src/CozyPouchLink.js

```diff
@@ -97,13 +105,38 @@ class PouchLink extends CozyLink {
     this.client = client
   }
 
+  async migrateAdapter({ fromAdapter, toAdapter, url, plugins }) {
+    try {
+      for (const plugin of plugins) {
+        PouchDB.plugin(plugin)
+      }
+      const doctypes = getPersistedSyncedDoctypes()
+      for (const doctype of doctypes) {
+        const prefix = getPrefix(url)
+        const dbName = getDatabaseName(prefix, doctype)
+        const oldPouch = new PouchDB(dbName, {
+          adapter: fromAdapter,
+          location: 'default'
+        })
+        await migratePouch(oldPouch, toAdapter)
+        await oldPouch.close()
+        await oldPouch.destroy()
```

Comment:
In order to be collected by the garbage, we need to "re init" the oldPouch var. It can be done by writing `oldPouch = null" or by using weakref (https://web.dev/detached-window-memory-leaks/)

Reply:
Quite interesting, thanks!

Reply:
Fixed with 78d3d78